clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/CloudMetricProvider.groovy

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.model
+
+interface CloudMetricProvider<T extends CloudMetric> {
+
+  /**
+   * Returns the platform of the provider
+   * @return a String, e.g. 'aws', 'gcp'
+   */
+  String getCloudProvider()
+
+  /**
+   * Returns a specific metric
+   * @param account the account
+   * @param region the region
+   * @param filters a collection of identifiers used to uniquely identify a metric
+   * @return a metric if one is found; should throw an exception if multiple metrics are found for the supplied filters
+   */
+  T getMetric(String account, String region, Map<String, String> filters)
+
+  /**
+   * Returns a list of metrics matching the supplied filters
+   * @param account the account
+   * @param region the region
+   * @param filters a collection of identifiers used to select a subset of all metrics in the account and region
+   * @return a list of metrics matching the filters
+   */
+  List<T> listMetrics(String account, String region, Map<String, String> filters)
+
+  /**
+   * Returns a statistic set for the metric uniquely identified by the supplied filters
+   * @param account the account
+   * @param region the region
+   * @param filters a collection of identifiers used to uniquely identify a metric
+   * @return a CloudMetricStatistics object, describing the statistics with timestamps
+   */
+  CloudMetricStatistics getStatistics(String account, String region, Map<String, String> filters)
```

Comment:
Do you want a time bound here? Probably dont want from the beginning of time and are interested in only some span of recent history. Perhaps this should have a start time or relative time offset from now (e.g. last 5 minutes).